src/sql/workbench/parts/notebook/models/notebookModel.ts

```diff
@@ -73,6 +73,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	private _oldKernel: nb.IKernel;
 	private _clientSessionListeners: IDisposable[] = [];
 	private _connectionUrisToDispose: string[] = [];
+	private _connectionsForAttachTo: Map<string, ConnectionProfile> = new Map<string, ConnectionProfile>();
```

Comment:
We already have _otherConnections and _defaultConnection in the model. Do we need to introduce another object that keeps track of connections?